x-pack/solutions/observability/plugins/apm/public/components/shared/focused_trace_waterfall/index.tsx

```diff
@@ -0,0 +1,180 @@
+
+    return [waterfallItem, ...convertedChildren];
+  }
+
+  return children.flatMap(convert);
+}
+
+const calculateOffset = ({
+  item,
+  rootWaterfallTransaction,
+}: {
+  item: IWaterfallTransaction | IWaterfallSpan;
+  rootWaterfallTransaction: IWaterfallTransaction;
+}) => item.doc.timestamp.us - rootWaterfallTransaction.doc.timestamp.us;
+
+export function FocusedTraceWaterfall({ items, isEmbeddable = false }: Props) {
+  const { euiTheme } = useEuiTheme();
+
+  const traceItems = items.traceItems;
+
+  const waterfall: {
+    items: Array<IWaterfallTransaction | IWaterfallSpan>;
+    totalDuration: number;
+    focusedItemId?: string;
+  } = useMemo(() => {
+    const waterfallItems: Array<IWaterfallTransaction | IWaterfallSpan> = [];
+
+    if (!traceItems) {
+      return {
+        items: [],
+        totalDuration: 0,
+        focusedItemId: undefined,
+      };
+    }
+
+    const rootWaterfallTransaction = getTransactionItem(
+      traceItems.rootTransaction as WaterfallTransaction,
+      0
+    );
+
+    waterfallItems.push(rootWaterfallTransaction);
+
+    const parentItem = traceItems.parentDoc
+      ? traceItems.parentDoc.processor.event === 'transaction'
+        ? getTransactionItem(traceItems.parentDoc as WaterfallTransaction, 0)
+        : getSpanItem(traceItems.parentDoc as WaterfallSpan, 0)
+      : undefined;
+
+    if (parentItem && parentItem.id !== rootWaterfallTransaction.id) {
+      parentItem.offset = calculateOffset({ item: parentItem, rootWaterfallTransaction });
+      waterfallItems.push(parentItem);
+    }
+
+    const focusedItem =
+      traceItems.focusedTraceDoc.processor.event === 'transaction'
+        ? getTransactionItem(traceItems.focusedTraceDoc as WaterfallTransaction, 0)
+        : getSpanItem(traceItems.focusedTraceDoc as WaterfallSpan, 0);
+
+    focusedItem.offset = calculateOffset({ item: focusedItem, rootWaterfallTransaction });
+
+    if (focusedItem.id !== rootWaterfallTransaction.id) {
+      waterfallItems.push(focusedItem);
+    }
+
+    const focusedItemChildren = convertChildrenToWatefallItem(
+      traceItems.focusedTraceDocChildren,
+      rootWaterfallTransaction
+    );
+
+    waterfallItems.push(...focusedItemChildren);
+
+    getLegendsAndColorBy(waterfallItems);
+
+    return {
+      items: waterfallItems,
+      totalDuration: rootWaterfallTransaction.duration,
+      focusedItemId: focusedItem.id,
+    };
+  }, [traceItems]);
+
+  if (!waterfall.items.length) {
+    return null;
+  }
+
+  return (
+    <>
+      <div
+        css={css`
+          position: relative;
+        `}
+      >
+        <div
+          css={css`
+            display: flex;
+            position: sticky;
+            top: var(--euiFixedHeadersOffset, 0);
+            z-index: ${euiTheme.levels.menu};
+            background-color: ${euiTheme.colors.emptyShade};
+            border-bottom: 1px solid ${euiTheme.colors.mediumShade};
```

Comment:
is this the same `euiTheme.border.thin`?

Reply:
yes, it is.. thanks.